docker/docker_client.go

```diff
@@ -187,27 +187,31 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 	}, nil
 }
 
+type requestOptions struct {
+	remoteName string
+	actions    string
+}
```

Comment:
`requestOptions` feels rather generic. Call this `authScope`, or, perhaps preferably, just use 2 string parameters?